### ISSUE TITLE
minetest.place_schematic fails to rotate nodes with facedir values > 3

### ISSUE THREAD

**Issue by t4im** [minetest/minetest#3093]
Schematics placed with
```lua
minetest.place_schematic(pos, schematic, rotation, replacements, force_placement)
```

with a rotation other than "0" will correctly rotate the first 4 facedir states (0-3), but will skip any other.

Tested with ffe3f03
[shakedown](https://github.com/t4im/shakedown)-unittest: [schematics](https://github.com/t4im/coretest/blob/master/schematics.lua)

**Comment by paramat** [comment]
This was requested by Sokomine a while ago. Not so much a bug more a feature request, as schematics were never meant to cope with complex rotations. Implementing this is so complex as to be not worth the effort according to hmmmm. I almost started working on this myself but then realised the complexity.

**Comment by t4im** [comment]
This is not about support rotations other than around the Y axis (though that would indeed be a very nice feature to have)
It's about the already supported rotation failing for certain facedir values.
Worst case scenario is perhaps a nasty switch construct or lookup table. Thats a maximum of 24 mappings. Less, if theres some system behind facedir assignment.

**Comment by paramat** [comment]
I wasn't thinking straight earlier. Yes this needs fixing. Certainly schematic rotations will remain just the 4 around the Y axis.

**Comment by Sokomine** [comment]
Fixing would be a good idea. It's one of the reasons why I have to spawn the schematics manually. Mirroring along one axis would also be great, although that's very tricky and does not work with all shapes.

**Comment by paramat** [comment]
See #3194 needs testing.

**Comment by paramat** [comment]
Merged a56aedb4eae7d21864ab8dc56a82f644c4639f7a
